### PR TITLE
Fix JSON schema compatibility bugs in adding/removing items in tuple

### PR DESCRIPTION
**Pull request by hhkkxxx133**
#1347

### CODE REVIEW

**Comment by EricWittmann** [comment]
Just waiting on a review from @jsenko for this, but LGTM.  Thanks @hhkkxxx133

**Review by jsenko — APPROVED** [review]
LGTM

**Comment by EricWittmann** [comment]
Thanks @hhkkxxx133 !

**Pull request merged**
